libpod/healthcheck.go

```diff
@@ -0,0 +1,62 @@
+package libpod
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+// HealthCheckStats represents the current state of a container
+type HealthCheckStatus int
+
+const (
+	HealthCheckSuccess           HealthCheckStatus = 0
+	HealthCheckFailure           HealthCheckStatus = 1
+	HealthCheckContainerStopped  HealthCheckStatus = 99
+	HealthCheckContainerNotFound HealthCheckStatus = 999
+	HealthCheckNotDefined        HealthCheckStatus = 9999
+)
+
+type HealthCheck struct {
+	interval    int //(default: 30s)
+	timeout     int //(default: 30s)
+	startPeriod int //(default: 0s)
+	retries     int //(default: 3)
+	command     string
+}
+
+// HealthCheck executes the ...
+func (r *Runtime) HealthCheck(name string) (HealthCheckStatus, error) {
```

Comment:
Can we make this a container method instead? It seems more natural to do container.HealthCheck()

Reply:
Oh, there's one of those below... Then why do we need this?

Reply:
Still wondering why we need this? It's just a LookupContainer and RunHealthCheck combined

Reply:
I want to expose two API endpoints based on "entry".  One is a runtime and one is a container.  I believe it will be needed in one of the next health check phases.  If not, I will cull it.